mvapich/default.nix

```diff
@@ -1,10 +1,29 @@
 { lib, stdenv, fetchurl, pkgconfig, bison, numactl, libxml2
-, perl, gfortran, slurm, openssh, hwloc, rdma-core
-, infiniband-diags, opensm, zlib
+, perl, gfortran, slurm, openssh, hwloc, zlib, makeWrapper
+# InfiniBand dependencies
+, infiniband-diags, opensm, rdma-core
+# OmniPath dependencies
+, libpsm2, libfabric
 # Compile with slurm as a process manager
 , useSlurm ? false
+# Network type for MVAPICH2
+, network ? "ethernet"
 } :
 
+assert builtins.elem network [ "ethernet" "infiniband" "omnipath" ];
+
+assert
+  (if network == "infiniband"
+     then (lib.lists.all (x: x != null) [ infiniband-diags opensm rdma-core ])
```

Comment:
I don't think this is needed. We always hand over these input parameters and non-null values. This assert and the next one can be removed.